app/controllers/api/automate_controller.rb

```diff
@@ -1,19 +1,33 @@
 module Api
   class AutomateController < BaseController
+    def index
+      resources = fetch_resources
+      render_resource :automate, :name => "automate", :subcount => resources.count, :resources => resources
+    end
+
     def show
-      ae_browser = MiqAeBrowser.new(User.current_user)
-      begin
-        resources = ae_browser.search(@req.c_suffix, ae_search_options)
-      rescue => err
-        raise BadRequestError, err.to_s
-      end
-      attributes = params['attributes'] ? %w(fqname) | params['attributes'].to_s.split(',') : nil
-      resources = resources.collect { |resource| resource.slice(*attributes) } if attributes.present?
+      resources = fetch_resources(@req.c_suffix)
       render_resource :automate, :name => "automate", :subcount => resources.count, :resources => resources
     end
 
     private
 
+    def fetch_resources(object_ref = nil)
+      resources = ae_browser.search(object_ref, ae_search_options)
+      resources.collect! { |resource| resource.slice(*attribute_params) } if attribute_params.present?
+      resources
+    rescue => err
+      raise BadRequestError, err.to_s
+    end
+
+    def ae_browser
+      @ae_browser ||= MiqAeBrowser.new(User.current_user)
```

Comment:
I woudn't memoize this. Recreate it with each request (User.current_user may be different).